sanic/app.py

```diff
@@ -1552,10 +1552,19 @@ def _loop_add_task(
         name: Optional[str] = None,
         register: bool = True,
     ) -> Task:
-        prepped = cls._prep_task(task, app, loop)
-        task = loop.create_task(prepped, name=name)
+        if not isinstance(task, Future):
+            prepped = cls._prep_task(task, app, loop)
+            if sys.version_info == (3, 7):
+                if name:
+                    error_logger.warning(
+                        "Cannot set a name for a task when using Python 3.7. Your "
+                        "task will be created without a name."
+                    )
+                task = loop.create_task(prepped)
+            else:
+                task = loop.create_task(prepped, name=name)
 
-        if name and register:
+        if name and register and sys.version_info > (3, 7):
```

Comment:
Again to keep the version checks consistent
```suggestion
        if name and register and sys.version_info >= (3, 8):
```

Reply:
Not sure if this check will work because of what we've just spoken on Discord (tuple len is 3)